bundles/org.smarthomej.binding.androiddebugbridge/src/main/java/org/smarthomej/binding/androiddebugbridge/internal/AndroidDebugBridgeDevice.java

```diff
@@ -118,47 +126,62 @@ public String getCurrentPackage() throws AndroidDebugBridgeDeviceException, Inte
             if (packageActivityName.contains("/"))
                 return packageActivityName.split("/")[0];
         }
-        throw new AndroidDebugBridgeDeviceReadException("Unable to read package name");
+        throw new AndroidDebugBridgeDeviceReadException(
+                "Device does not support " + CURRENT_PACKAGE_CHANNEL + ": " + out);
     }
 
-    public boolean isAwake()
-            throws InterruptedException, AndroidDebugBridgeDeviceException, TimeoutException, ExecutionException {
-        String devicesResp = runAdbShell("dumpsys", "activity", "|", "grep", "mWakefulness");
-        return devicesResp.contains("mWakefulness=Awake");
+    public @Nullable Boolean isAwake() throws InterruptedException, AndroidDebugBridgeDeviceException,
```

Comment:
```suggestion
    public Optional<Boolean> isAwake() throws InterruptedException, AndroidDebugBridgeDeviceException,
```

Reply:
did but do not understand why to make it more complicated with optional instead to use null as a status